server/src/main/java/org/opensearch/action/admin/indices/get/GetIndexResponse.java

```diff
@@ -105,15 +99,23 @@ public GetIndexResponse(
         this.indices = in.readStringArray();
 
         int mappingsSize = in.readVInt();
-        ImmutableOpenMap.Builder<String, ImmutableOpenMap<String, MappingMetadata>> mappingsMapBuilder = ImmutableOpenMap.builder();
+        ImmutableOpenMap.Builder<String, MappingMetadata> mappingsMapBuilder = ImmutableOpenMap.builder();
         for (int i = 0; i < mappingsSize; i++) {
-            String key = in.readString();
-            int valueSize = in.readVInt();
-            ImmutableOpenMap.Builder<String, MappingMetadata> mappingEntryBuilder = ImmutableOpenMap.builder();
-            for (int j = 0; j < valueSize; j++) {
-                mappingEntryBuilder.put(in.readString(), new MappingMetadata(in));
+            String index = in.readString();
+            if (in.getVersion().before(Version.V_2_0_0)) {
+                int numMappings = in.readVInt();
+                assert numMappings == 0 || numMappings == 1 : "Expected 0 or 1 mappings but got " + numMappings;
+                if (numMappings == 1) {
+                    String type = in.readString();
```

Comment:
I think we should try to handle the `numMappings` a bit more gracefully, generalizing for `numMappings > 0`
```
if (numMappings > 0) {
  for (int j = 0; j < numMappings; j++) {
      String type = in.readString();
      ...
  }
|
```

Assertions could be turned off, so we would fail with unconsumed input stream after (potentially).

Reply:
But there can't be more than 1 mapping. Which is why there's the assertion.